src/python/pants/engine/fs.py

```diff
@@ -3,10 +3,12 @@
 
 from dataclasses import dataclass
 from enum import Enum
-from typing import TYPE_CHECKING, Iterable, Optional, Tuple
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, cast
 
 from pants.engine.collection import Collection
-from pants.engine.rules import RootRule, side_effecting
+from pants.engine.rules import RootRule, collect_rules, rule, side_effecting
+from pants.option.global_options import GlobalOptions
+from pants.option.global_options import GlobMatchErrorBehavior as GlobMatchErrorBehavior
```

Comment:
We'd have a cycle with `global_options` if `GlobMatchErrorBehavior` was defined here. We re-export it so that other files can import it from `fs.py`.